### PR TITLE
Use short type var names

### LINKED ISSUE #1872: Follow PEP 8 recommendations for type variable names
This appeared in https://github.com/python/typeshed/pull/1856

The problem is that type variable names in some stubs don't follow PEP 8 that recommends short names. This recommendation exists for a reason: behaviour of some constructs is subtly different with type variables and normal types, so it should be immediately clear that a given name refers to a type variable. Most notable examples are generic base classes and generic type aliases, where this leads to false negatives (unless `--disallow-any-generics` is used):
```python
LongNameLikeAClass = TypeVar('LongNameLikeAClass')
# 300 lines later
class Cls(Iterable[LongNameLikeAClass]):  # it's not obvious this class is generic
    ...
Alias = Dict[str, List[LongNameLikeAClass]]  # even more here, this looks like a normal alias

def fun(arg: Cls) -> Alias:  # two `Any`s sneaked in here
    ...
```
(it is not something hypothetic, I have seen this kind of errors many times)

I propose to replace the existing long type variable names to follow PEP 8 style. If someone really doesn't like short names, we could propose some other agreement, like end long type variable names with `_TVar`.

### PR DESCRIPTION
**Pull request by srittau**
Closes #1872

### CODE REVIEW

File: stdlib/2/__builtin__.pyi [hunk 1]
```diff
@@ -1440,12 +1440,12 @@ Ellipsis: ellipsis
...
-    _AnyBuffer = TypeVar('_AnyBuffer', str, unicode, bytearray, buffer)
+    _ABT = TypeVar('_ABT', str, unicode, bytearray, buffer)
 
     class buffer(Sized):
-        def __init__(self, object: _AnyBuffer, offset: int = ..., size: int = ...) -> None: ...
-        def __add__(self, other: _AnyBuffer) -> str: ...
-        def __cmp__(self, other: _AnyBuffer) -> bool: ...
+        def __init__(self, object: _ABT, offset: int = ..., size: int = ...) -> None: ...
+        def __add__(self, other: _ABT) -> str: ...
+        def __cmp__(self, other: _ABT) -> bool: ...
```
Comment (ilevkivskyi) [inline stdlib/2/__builtin__.pyi]: Should these be just unions?

**Comment by JelleZijlstra** [comment]
I'm not sure I agree with this. Short variable names are generally bad for readability and I don't know why typevars are special. For example, isn't `_AnyStr2` clearer than `_AS`?

File: stdlib/2/__builtin__.pyi (hunk 1 above)
Comment (srittau) [reply stdlib/2/__builtin__.pyi]: Good point. I changed it to a union and changed the name back to be more expressive.

**Comment by srittau** [comment]
@JelleZijlstra Personally, I agree. I prefer a solution like always appending `_var`, `_co`, and `_contra` or ending all TypeVars with `T`. But PEP 8 recommends something different at the moment. Maybe we should send this back to #1872 and possibly get the recommendation in PEP 8 changed?

**Comment by srittau** [comment]
I am going to close this for now, until we decide what to do in #1872 .

**Pull request closed without merging**